### PR TITLE
Show flows in progress when picking a handler

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/config-flow/dialog-data-entry-flow.ts [hunk 1]
```diff
@@ -84,59 +91,89 @@ class DataEntryFlowDialog extends LitElement {
...
+        step = await params.flowConfig.fetchFlow(
+          this.hass,
+          params.continueFlowId
+        );
+      } catch (err) {
+        this._step = undefined;
+        this._params = undefined;
+        showAlertDialog(this, {
+          title: "Error",
+          text: "Config flow could not be loaded",
```
Comment (ludeeus) [inline src/dialogs/config-flow/dialog-data-entry-flow.ts:113]: Should use localize

File: src/dialogs/config-flow/dialog-data-entry-flow.ts [hunk 2]
```diff
@@ -84,59 +91,89 @@ class DataEntryFlowDialog extends LitElement {
...
+      this._loading = false;
       return;
     }
 
-    // Happens if second showDialog called
-    if (curInstance !== this._instance) {
-      return;
+    // Create a new config flow. Show picker
+    if (!params.flowConfig.getFlowHandlers) {
+      throw new Error("No getFlowHandlers defined in flow config");
```
Comment (ludeeus) [inline src/dialogs/config-flow/dialog-data-entry-flow.ts:130]: Why throw an Error and not showAlertDialog?

File: src/dialogs/config-flow/dialog-data-entry-flow.ts [hunk 3]
```diff
@@ -291,6 +337,39 @@ class DataEntryFlowDialog extends LitElement {
...
+    if (!flowsInProgress.length) {
+      let step: DataEntryFlowStep;
+      try {
+        step = await this._params!.flowConfig.createFlow(this.hass, handler);
+      } catch (err) {
+        this._step = undefined;
+        this._params = undefined;
+        showAlertDialog(this, {
+          title: "Error",
+          text: "Config flow could not be loaded",
```
Comment (ludeeus) [inline src/dialogs/config-flow/dialog-data-entry-flow.ts:356]: Should use localize

File: src/dialogs/config-flow/step-flow-pick-flow.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false })
+  public flowsInProgress!: DataEntryFlowProgress[];
+
+  @property() public handler!: string;
+
+  protected render(): TemplateResult {
+    return html`
+      <h2>We discovered these, want to set them up?</h2>
```
Comment (ludeeus) [inline src/dialogs/config-flow/step-flow-pick-flow.ts:36]: Should use localize

File: src/dialogs/config-flow/step-flow-pick-flow.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+            <paper-item-body>
+              ${localizeConfigFlowTitle(this.hass.localize, flow)}
+            </paper-item-body>
+            <ha-icon-next></ha-icon-next>
+          </paper-icon-item>`
+        )}
+        <paper-item @click=${this._startNewFlowPicked} .handler=${this.handler}>
+          <paper-item-body>
+            No, set up an other instance of
```
Comment (ludeeus) [inline src/dialogs/config-flow/step-flow-pick-flow.ts:59]: Should use localize

**Review by ludeeus — APPROVED** [review]

**Comment by ludeeus** [comment]
Nice one!

File: src/dialogs/config-flow/dialog-data-entry-flow.ts (hunk 2 above)
Comment (bramkragten) [reply src/dialogs/config-flow/dialog-data-entry-flow.ts:130]: This is not a user error

**Pull request merged**